capstone/src/main/java/com/google/sps/servlets/ChallengeServlet.java

```diff
@@ -27,14 +32,26 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     Query query = new Query("Challenge").addSort("timestamp", SortDirection.DESCENDING);
     PreparedQuery results = datastore.prepare(query);
+    UserService userService = UserServiceFactory.getUserService();
+    String userId = "";
+    if (userService.isUserLoggedIn()) {
+      userId = userService.getCurrentUser().getUserId();
+    } else {
+      ErrorHandler.sendError(response, "User is not logged in.");
```

Comment:
Have you tested this manually? I had a brief back and forth with @himaniyadav about whether or not you should return after sending an error, and it looks like in this case you would still be sending a challenge response even though the user isn't logged in.

My hunch is you want to return after sending an error, but make sure that the client-side also handles whatever errors you're sending in a reasonable way.

Reply:
Yeah I also noticed that yesterday. Anika, Himani and I discussed this and I think we are going to send an alert on the client side and just get rid of ErrorHandler